blocks/dialog/dialog.js

```diff
@@ -0,0 +1,69 @@
+import { loadFragment } from '../fragment/fragment.js';
+import { decorateButtons, decorateIcons } from '../../scripts/aem.js';
+
+function getCell(table, key) {
+  const resultRow = Array.from(table.children)
+    .find((row) => row.firstElementChild.textContent.toLowerCase() === key.toLowerCase());
+  return resultRow?.lastElementChild;
+}
+
+function formatButton(button) {
+  if (button.classList.contains('button-container')) return button;
+
+  const buttonContainer = document.createElement('div');
+  buttonContainer.classList.add('button-container');
+  buttonContainer.innerHTML = '<a href="#" class="button"></a></div>';
+  buttonContainer.querySelector('a').textContent = button.textContent;
+  return buttonContainer;
+}
+
+export function createDialog() {
+  const dialog = document.createElement('dialog');
+
+  const closeButton = document.createElement('button');
+  closeButton.classList.add('close-button');
+  closeButton.setAttribute('aria-label', 'Close');
+  closeButton.type = 'button';
+  closeButton.innerHTML = '<span class="icon icon-close"></span>';
```

Comment:
in lieu of an svg, i'm inclined to reuse the css implementation of the close icon in header

Reply:
IIUC, we would have to duplicate the code to use it here. And I find the 15lines of CSS have more complexity than the SVG inclusion.